calliope/core/preprocess/locations.py

```diff
@@ -397,15 +397,27 @@ def process_per_distance_constraints(tech_name, tech_settings, locations, locati
     return tech_settings
 
 
-def compute_depreciation_rates(tech_id, tech_config, warnings, errors):
+def check_costs_and_compute_depreciation_rates(tech_id, loc_or_link, tech_config, warnings, errors):
     cost_classes = list(tech_config.get('costs', {}).keys())
     for cost in cost_classes:
-        # If the cost class is empty, delete it now and warn about it
+
+        # Warning if a cost is defined without a cost class, which is probably a mistake
+        if cost in POSSIBLE_COSTS:
+            warnings.append(
+                '`{}` at `{}` defines {} as a cost class. '
+                'This is probably an indentation mistake.'.format(
+                    tech_id, loc_or_link, cost)
+            )
+
+        # Warning if a cost class is empty
         if not isinstance(tech_config.costs[cost], dict):
             warnings.append(
-                'Deleting empty cost class `{}` for technology `{}`.'.format(cost, tech_id)
+                'Deleting empty cost class `{}` for technology `{}` at `{}`.'.format(
+                    cost, tech_id, loc_or_link)
             )
             tech_config.costs.del_key(cost)
+            # If the cost class is empty, it is not enough to delete it,
```

Comment:
This comment isn't quite right, it's skipping the rest of the loop because it has deleted it (the rest of the loop expects the cost class to exist...)